data.md

```diff
@@ -1073,20 +1073,34 @@ Merkle Patricia Tree
       => #merkleBrancher ( M, BRANCHVALUE, PATH[0], PATH[1 .. #sizeByteArray(PATH) -Int 1], VALUE ) [owise]
 ```
 
+- `MerkleUpdateMap` Takes a mapping of `ByteArray |-> String` and generates a trie
+
+```k
+    syntax MerkleTree ::= MerkleUpdateMap( MerkleTree, Map ) [function]
+ // -------------------------------------------------------------------
+    rule MerkleUpdateMap( TREE, KEY |-> VALUE M ) => MerkleUpdateMap( MerkleUpdate( TREE, #nibbleize(KEY), VALUE ) , M )
+      requires M =/=K .Map
+
+    rule MerkleUpdateMap( TREE, KEY |-> VALUE ) => MerkleUpdate( TREE, #nibbleize(KEY), VALUE )
+
+    rule MerkleUpdateMap( TREE, .Map ) => TREE
```

Comment:
I can't see why we need a separate case for the singleton map `KEY |-> VALUE` and `KEY |-> VALUE M requires M =/=K .Map`? It looks like if you remove the side condition `M =/=K .Map` and remove the second rule, you get the same result as this current approach?